test/cppsuite/test_harness/workload_generator.h

```diff
@@ -149,20 +149,24 @@ class workload_generator : public component {
     static void
     execute_operation(thread_context &context)
     {
+        WT_SESSION *session;
         thread_operation operation;
 
-        // Set up work
-        operation = context.get_thread_operation();
+        session = connection::instance().get_session();
 
-        // Main loop
+        /* Main loop for worker threads. */
         while (context.is_running()) {
-            switch (operation) {
+            switch (context.get_thread_operation()) {
```

Comment:
It should be optimised, but try to avoid to call functions in a while loop. If not optimised, it will add overhead and cause perf issues.

Reply:
What if we expose the thread operation as a public const variable?

Reply:
We could, but all other variables are private and have getters, seems inconsistent. I'd usually avoid public variables to keep encapsulation.